metarecord/tests/test_function_model.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+from metarecord.models import Classification, Function
+
+
+@pytest.mark.django_db
+def test_function_get_name(function):
+    assert function.get_name() == function.classification.title
+
+    function.is_template = True
+    function.save(update_fields=("is_template",))
+
+    assert function.get_name() == function.name
+
+
+@pytest.mark.django_db
+def test_function_draft_delete(function):
+    function.state = Function.APPROVED
+    can_delete = function.can_user_delete(None)
+    assert can_delete is False
+
+
+@pytest.mark.django_db
+def test_function_save_without_classification(function):
+    function.classification = None
+    try:
+        function.save()
+    except Exception as e:
+        assert str(e) == "Classification is required."
+    else:
+        assert False, "Function save should have failed!"
```

Comment:
You could use the `pytest.raises` for this, since that's the pytest helper meant for checking the raised exceptions:
```python
with pytest.raises(Exception) as excinfo:
    function.save()
assert str(excinfo.value) == "Classification is required."
```

Also, does it really raise just bare Exception? Test should specify the exact exception type (usually it shouldn't Exception, but some subclass of it).

Reply:
The save method actually raises just a general `Exception` :smile: Should we change the Exception in the code or leave it as is?

Reply:
It could be changed to ValueError, which IMHO is more appropriate. (ValueError means that an invalid value is passed to function (compare to TypeError), but in this case it's not really passed, but given implicitly via self... I would still say it's close enough.)

Reply:
Can be done later in another PR.